build/.azure-pipelines.TemplateValidation.yml

```diff
@@ -4,79 +4,209 @@ jobs:
+        templateArgs2: '-preset=recommended -auth=custom'
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      CustomAuthMvvm_And_WebAuthMvux:
+        templateArgs1: '-preset=recommended -presentation mvvm -auth=custom'
+        templateArgs2: '-preset=recommended -auth=web'
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      WebAuthMvvm_And_OidcAuthMvux:
+        templateArgs1: '-preset=recommended -presentation mvvm -auth=web'
+        templateArgs2: '-preset=recommended -auth=oidc'
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      OidcAuthMvvm_And_MsalAuthMvux:
+        templateArgs1: '-preset=recommended -presentation mvvm -auth=oidc'
+        templateArgs2: '-preset=recommended -auth=msal'
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      MsalAuthMvvm_And_MauiBlank:
+        templateArgs1: '-preset=recommended -presentation mvvm -auth=msal'
+        templateArgs2: '-preset=blank -maui'
+        net7_workloads: 'android ios maccatalyst maui'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      # Combining MauiRecommended and MauiRecommendedNet8 is causing build failures in CI:
+      # C:\hostedtoolcache\windows\dotnet\sdk\8.0.100-rc.1.23463.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(485,5): error NETSDK1112: The runtime pack for Microsoft.Maui.Core.Runtime.android was not downloaded. Try running a NuGet restore with the RuntimeIdentifier 'android'. [D:\a\1\s\UnoApp1\UnoApp1.Mobile\UnoApp1.Mobile.csproj::TargetFramework=net7.0-android]
+      # They are split for now. When adding a new validation, try to plug it in one of these two PLACEHOLDERs
       MauiRecommended:
-       templateArgs: '-preset=recommended -maui'
+        templateArgs1: '-preset=recommended -maui'
+        templateArgs2: 'PLACEHOLDER'
+        net7_workloads: 'android ios maccatalyst maui'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
       MauiRecommendedNet8:
-       templateArgs: '-preset=recommended -maui -tfm net8.0'
+        templateArgs1: '-preset=recommended -maui -tfm net8.0'
+        templateArgs2: 'PLACEHOLDER'
+        net7_workloads: ''
+        net8_workloads: 'android ios maccatalyst maui'
+        install_net7: false
+        install_net8: true
 
   variables:
     - name: UseDotNetNativeToolchain
       value: false
 
   steps:
+  - checkout: none
+
+  - task: DownloadBuildArtifacts@0
+    inputs:
+      artifactName: $(Build.DefinitionName)
+
+  - script: |
```

Comment:
Why are we doing this step, since in the next step we're installing the template from the artifactsdirectory, rather than from where we've copied them to

Reply:
Ah, I don't know. This was an already existing step. I guess it's redundant. So let me try to clean it up in this PR.